### PR TITLE
Fix public activity showing private repos

### LINKED ISSUE #811: Public activity shows private repositories
- Gitea version (or commit ref): 1.0.1
- Git version: 2.7.4
- Operating system: Ubuntu 16.04.1
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes: https://try.gitea.io/publicActivity?tab=activity
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description
After making a public repository private, its activity (creation of repo, pushed commits, etc.) is still visible in the Public Activity of the user. See https://try.gitea.io/publicActivity?tab=activity for an example. The links to the commits and repository all result in 404s, but anyone can still read the name of the private repository and the titles of the pushed commits.

### PR DESCRIPTION
**Pull request by thehowl**
Fixes #811

### CODE REVIEW

File: models/repo.go [hunk 1]
```diff
@@ -1408,6 +1408,13 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
 			}
 		}
 
+		// If repo has become private, we need to set its actions to private.
+		if repo.IsPrivate {
+			e.Where("repo_id = ?", repo.ID).Cols("is_private").Update(&Action{
```
Comment (ethantkoenig) [inline models/repo.go]: We should check the returned `error`

File: models/repo.go [hunk 2]
```diff
@@ -1408,6 +1408,13 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
...
+				IsPrivate: true,
+			})
+		}
```
Comment (ethantkoenig) [inline models/repo.go]: Shouldn't we also mark the actions as public if the repo becomes public?

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

File: models/repo.go (hunk 2 above)
Comment (thehowl) [reply models/repo.go]: I've thought about that, and I am not sure whether having a feed that adds new events "in the past" would make much sense.

Comment (ethantkoenig) [reply models/repo.go]: Fair enough

**Comment by ethantkoenig** [comment]
LGTM

**Comment by strk** [comment]
It would be good to add tests for bugfixes, had you looked at what it would take ?

File: models/repo.go [hunk 3]
```diff
@@ -1408,6 +1408,16 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
 			}
 		}
 
+		// If repo has become private, we need to set its actions to private.
+		if repo.IsPrivate {
+			_, err = e.Where("repo_id = ?", repo.ID).Cols("is_private").Update(&Action{
+				IsPrivate: true,
```
Comment (lunny) [inline models/repo.go:1414]: `IsPrivate: repo.IsPrivate`, ? And remove the `if repo.IsPrivate` ?

**Review by lunny — CHANGES_REQUESTED** [review]

File: models/repo.go (hunk 3 above)
Comment (thehowl) [reply models/repo.go:1414]: https://github.com/go-gitea/gitea/pull/892#discussion_r100639602

Comment (lunny) [reply models/repo.go:1414]: ?

Comment (thehowl) [reply models/repo.go:1414]: We already discussed this in the linked comment. As I said, "I am not sure whether having a feed that adds new events "in the past" would make much sense.".

**Comment by thehowl** [comment]
@strk, will look into tests later today

File: models/repo.go (hunk 3 above)
Comment (lunny) [reply models/repo.go:1414]: OK.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Comment by thehowl** [comment]
Added a test, as requested by @strk.

**Comment by strk** [comment]
LGTM, thanks for the test !

**Pull request merged**